File: src/main/java/com/manning/simplysend/repositories/UserRepository.java
package com.manning.simplysend.repositories;

import com.manning.simplysend.entities.User;
import org.springframework.data.repository.PagingAndSortingRepository;

import java.util.Optional;

public interface UserRepository extends PagingAndSortingRepository<User, Long> {
    Optional<User> findByEmail(String email);
}

File: src/main/kotlin/com/manning/simplysend/repositories/UserRepository.kt
package com.manning.simplysend.repositories

import com.manning.simplysend.entities.User
import org.springframework.data.repository.PagingAndSortingRepository
import java.util.*

interface UserRepository : PagingAndSortingRepository<User, Long> {

    fun findByEmail(email: String): Optional<User>
}